webapp/src/components/link_preview/link_preview.jsx

```diff
@@ -0,0 +1,184 @@
+import {GitMergeIcon, GitPullRequestIcon, IssueClosedIcon, IssueOpenedIcon, SkipIcon} from '@primer/octicons-react';
+import PropTypes from 'prop-types';
+import React, {useEffect, useState} from 'react';
+import ReactMarkdown from 'react-markdown';
+import './preview.css';
+
+import Client from 'client';
+import {getLabelFontColor} from '../../utils/styles';
+import {isUrlCanPreview} from 'src/utils/github_utils';
+
+const maxTicketDescriptionLength = 160;
+
+export const LinkPreview = ({embed: {url}, connected}) => {
+    const [data, setData] = useState(null);
+    useEffect(() => {
+        const initData = async () => {
+            if (isUrlCanPreview(url)) {
+                const [owner, repo, type, number] = url.split('github.com/')[1].split('/');
+
+                let res;
+                switch (type) {
+                case 'issues':
+                    res = await Client.getIssue(owner, repo, number);
+                    break;
+                case 'pull':
+                    res = await Client.getPullRequest(owner, repo, number);
+                    break;
```

Comment:
We'll want to inject these as props instead of accessing the client directly in this component

Reply:
Also I wonder what happens here if the user is connected, but doesn't have access to a private repo being linked here

Reply:
It bypasses this and shows default embed preview like this.
<img width="724" alt="image" src="https://github.com/mattermost/mattermost-plugin-github/assets/26380261/f0eeed87-5f1e-49ef-be47-a9f8e7fbb99f">

Reply:
I'm not sure I understand how that works. It shows the fallback because this component returns `null` in that case?

Reply:
Yes, I think it caused by [this](https://github.com/Sn-Kinos/mattermost-plugin-github/blob/0dfcb288bf0e72dad68a78aec09de94fa7e49eb7/server/plugin/api.go#L1077)

Reply:
I meant the fact that the webapp handles this gracefully when the component ends up returning null. I don't think you can coordinate React components together like that